### DISCUSSION TITLE
How long a failed test was running?

### DISCUSSION THREAD

**Discussion by szmarczak** [avajs/ava#2687] [Ideas]
* What you're trying to do

I'm trying to check if https://github.com/szmarczak/http2-wrapper/runs/1895479368?check_suite_focus=true#step:4:231 was caused by too small timeout or a bug in my code. If it took ~500ms, then I need to increase the timeout. If it's lower than 400ms, it's a bug in my code.

* Why you can't use AVA for this

AVA has no such feature. It only displays time for tests that passed.

* And maybe how you think AVA could handle this

https://github.com/avajs/ava/blob/d8b1e89e6e8b88e1582cf02f4cf088c87c2e058f/lib/reporters/default.js#L620-L637

with

```diff
+		const duration = event.duration > this.durationThreshold ? colors.duration(' (' + prettyMs(event.duration) + ')') : '';
+
		if (event.type === 'hook-failed' || event.type === 'test-failed') {
			if (this.verbose) {
-				this.write(`${colors.error(figures.cross)} ${this.prefixTitle(event.testFile, event.title)} ${colors.error(event.err.message)}`);
+				this.write(`${colors.error(figures.cross)} ${this.prefixTitle(event.testFile, event.title)} ${colors.error(event.err.message)}${duration}`);
			} else {
				this.write(this.prefixTitle(event.testFile, event.title));
			}
		} else if (event.knownFailing) {
			if (this.verbose) {
-				this.write(`${colors.error(figures.tick)} ${colors.error(this.prefixTitle(event.testFile, event.title))}`);
+this.write(`${colors.error(figures.tick)} ${colors.error(this.prefixTitle(event.testFile, event.title))}${duration}`);
			} else {
				this.write(colors.error(this.prefixTitle(event.testFile, event.title)));
			}
		} else if (this.verbose) {
-			const duration = event.duration > this.durationThreshold ? colors.duration(' (' + prettyMs(event.duration) + ')') : '';
			this.write(`${colors.pass(figures.tick)} ${this.prefixTitle(event.testFile, event.title)}${duration}`);
		} else {
			this.write(this.prefixTitle(event.testFile, event.title));
		}
```

**Comment by novemberborn** [comment]
We've already decided to change under what circumstances these durations are printed, see https://github.com/avajs/ava/issues/1668. Your use case doesn't fall within those constraints.

I understand what you're trying to do here, but on the other hand, we can't provide you with all the tools you might need to debug a failing test. This is easily dealt with by adding debugging code.

**Reply by szmarczak** [reply]
After some thoughts I think I understand this move. #1668 is a very good point!
